src/components/dialog/dialog.pw.tsx

```diff
@@ -0,0 +1,596 @@
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent aria-describedby="aria description" />);
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-describedby",
+      "aria description"
+    );
+  });
+
+  test("when aria-labelledby prop is passed and title prop is not, the aria-labelledby attribute of the Dialog should be set to the prop's value", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <DialogComponent title={undefined} aria-labelledby="label-element-id" />
+    );
+    await expect(page.getByRole("dialog")).toHaveAttribute(
+      "aria-labelledby",
+      "label-element-id"
+    );
+  });
+
+  test("when disableClose prop is passed, close icon should be disabled", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent disableClose />);
+    await expect(page.getByLabel("Close")).toBeDisabled();
+  });
+
+  test("when help prop is provided, hovering over the rendered help icon displays help text", async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <DialogComponent title="Sample Dialog" help="Some help text" />
+    );
+
+    const helpIcon = page.getByLabel("help");
+    await helpIcon.hover();
+
+    await expect(page.getByRole("tooltip")).toHaveText("Some help text");
+  });
+
+  test("when Dialog is opened, the first focusable element should be focused", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogWithFirstFocusableElement />);
+    await expect(
+      page.getByRole("button").filter({ hasText: "Press me" })
+    ).toBeFocused();
+  });
+
+  test("when disableAutoFocus prop is passed, the first focusable element should not be focused", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogWithFirstFocusableElement disableAutoFocus />);
+    await expect(
+      page.getByRole("button").filter({ hasText: "Press me" })
+    ).not.toBeFocused();
+  });
+
+  test("when tabbing through Dialog content, focus should remain trapped inside the Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent />);
+    const dialog = page.getByRole("dialog");
+    const thirdTextbox = page.getByLabel("Textbox3");
+    const secondTextbox = page.getByLabel("Textbox2");
+    const firstTextbox = page.getByLabel("Textbox1");
+    const closeButton = page.getByLabel("Close");
+
+    await dialog.press("Tab");
+    await expect(closeButton).toBeFocused();
+
+    await closeButton.press("Tab");
+    await expect(firstTextbox).toBeFocused();
+
+    await firstTextbox.press("Tab");
+    await expect(secondTextbox).toBeFocused();
+
+    await secondTextbox.press("Tab");
+    await expect(thirdTextbox).toBeFocused();
+
+    await thirdTextbox.press("Tab");
+    await expect(closeButton).toBeFocused();
+  });
+
+  test("when shift tabbing through Dialog content, focus should remain trapped inside the Dialog", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DialogComponent />);
```

Comment:
suggestion: (non-blocking) it would be better to add a space to separate the mount from the assertion to follow the same pattern as with the other test cases.